### DISCUSSION TITLE
fromRequest not working

### DISCUSSION THREAD

**Discussion by mzanetteo2000** [spatie/laravel-data#554] [Q&A] [answered]
Hi, 
I have problem with fromRequest magic method, I use it to map the correct name of my param from the request to the data Class.

```
// controller
public function store(Request $request){
        try{

            $eventData = app(SongData::class);

        } catch(\Exception $e){
            dd($e->getMessage());
        }
    }
```

```
// data 
class SongData extends Data
{
    public function __construct(
        public string $title,
        public string $artist,
    ) {
    }

    public static function fromRequest(Request $request): static
    {
        return new self(
            $request->input('title_of_song'),
            $request->input('artist_name')
        );
    }

}
```

I always get a validation exception that tell me is missing title field, but I have mapped it in the formRequest with another name.

Am I forgetting something? I  also recreate the documentation example but it's not working.

**Comment by rubenvanassche** [comment, marked as answer]
At the moment validation error messages aren't using mapped names. We're aware of this and working on a fix in v4 since validation is completely rethought over there.

**Comment by mzanetteo2000** [comment]
Thanks for the reply. Have you already planned a possible release date for version 4?

**Reply by rubenvanassche** [reply]
Nowp we're completely rethinking some parts and that's taking a lot of time. Hopefully this year
